### PR TITLE
Wine build script update

### PR DESCRIPTION
**Pull request by PeterSurda**
- update and clean up and make sure it works
- it builds the binary but I haven't tried to run the binary
- it probably still missing some fine tuning, OpenCL probably doesn't work

### CODE REVIEW

**Comment by g1itch** [comment]
This is what I got on my Ubuntu 18 virtual machine:
```
FINISHED --2020-02-11 18:01:54--
Total wall clock time: 1m 39s
Downloaded: 5 files, 78M in 1m 33s (861 KB/s)
Setting up wine
Installing Python 2.7.17 64b
./buildscripts/winbuild.sh: line 62: wine: command not found
Installing vcredist for 64 bit
./buildscripts/winbuild.sh: line 64: wine: command not found
Upgrading pip
./buildscripts/winbuild.sh: line 66: wine: command not found
Installing PyQt-4-4.11.4-gpl-Py2.7-Qt4.8.7 64b
./buildscripts/winbuild.sh: line 84: wine: command not found
Installing OpenSSL 1_0_2t 64b
./buildscripts/winbuild.sh: line 94: wine: command not found
Installing PyOpenCL
./buildscripts/winbuild.sh: line 118: wine: command not found
Installing msgpack
./buildscripts/winbuild.sh: line 110: wine: command not found
Installing PyInstaller
./buildscripts/winbuild.sh: line 104: wine: command not found
Create dll
./buildscripts/winbuild.sh: line 130: x86_64-w64-mingw32-g++: command not found
./buildscripts/winbuild.sh: line 131: x86_64-w64-mingw32-g++: command not found
./buildscripts/winbuild.sh: line 142: wine: command not found

```

File: buildscripts/builder.sh [hunk 1]
```diff
@@ -1,173 +0,0 @@
...
-fi
-#Functions
-function install_wine {
-
-	
-	wget -nc https://dl.winehq.org/wine-builds/Release.key  --no-check-certificate
-	sudo apt-key add Release.key
-	sudo apt-add-repository 'https://dl.winehq.org/wine-builds/ubuntu/'
-	sudo apt-get -y update
-	sudo apt-get -y install wine1.8 winetricks
```
Comment (g1itch) [inline buildscripts/builder.sh]: That apt commands are probably needed

Comment (g1itch) [reply buildscripts/builder.sh]: Maybe `sudo apt-get -y install wine1.8 winetricks` is sufficient

**Comment by g1itch** [comment]
Now, after installing wine:
```
Traceback (most recent call last):
  File "c:\Python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "c:\Python27\lib\site-packages\PyInstaller\__main__.py", line 114, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\Python27\lib\site-packages\PyInstaller\__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\Python27\lib\site-packages\PyInstaller\building\build_main.py", line 734, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\Python27\lib\site-packages\PyInstaller\building\build_main.py", line 681, in build
    exec(code, spec_namespace)
  File "bitmessagemain.spec", line 97, in <module>
  File "c:\Python27\lib\site-packages\PyInstaller\building\api.py", line 436, in __init__
    upx_exclude=self.upx_exclude
  File "c:\Python27\lib\site-packages\PyInstaller\building\api.py", line 199, in __init__
    self.__postinit__()
  File "c:\Python27\lib\site-packages\PyInstaller\building\datastruct.py", line 160, in __postinit__
    self.assemble()
  File "c:\Python27\lib\site-packages\PyInstaller\building\api.py", line 262, in assemble
    dist_nm=inm)
  File "c:\Python27\lib\site-packages\PyInstaller\building\utils.py", line 220, in checkCache
    digest = cacheDigest(fnm, redirects)
  File "c:\Python27\lib\site-packages\PyInstaller\building\utils.py", line 365, in cacheDigest
    with open(fnm, "rb") as f:
IOError: [Errno 2] No such file or directory: 'Z:\\home\\glitch\\PyBitmessage\\src\\bitmsghash\\bitmsghash64.dll'

```

File: buildscripts/builder.sh [hunk 2]
```diff
@@ -1,173 +0,0 @@
...
-function build_dll(){
-	cd $BASE_DIR
-	rm -rf master.zip
-	rm -rf PyBitmessage
-	git clone https://github.com/Bitmessage/PyBitmessage.git
-	cd PyBitmessage/src/bitmsghash
-	if [ ${MACHINE_TYPE} == 'x86_64' ]; then
-		# Do stuff for 64 bit machine
-		echo "Install MinGW"
-		sudo apt-get -y install mingw-w64
```
Comment (g1itch) [inline buildscripts/builder.sh]: `sudo apt-get -y install mingw-w64` seems to be needed also

**Comment by g1itch** [comment]
OK, it was built. But in ` ~/.wine64/dosdevices/z\:/home/glitch/PyBitmessage/packages/pyinstaller/dist/`. Maybe copy the executable into build dir?

**Comment by g1itch** [comment]
The executable starts, but if I go to "Settings" the exception appears:
```
2020-02-11 18:37:44,121 - CRITICAL - Unhandled exception
Traceback (most recent call last):
  File "bitmessageqt\__init__.py", line 2466, in click_actionSettings
  File "bitmessageqt\settings.py", line 42, in __init__
  File "bitmessageqt\widgets.py", line 13, in load
  File "site-packages\PyQt4\uic\__init__.py", line 236, in loadUi
  File "site-packages\PyQt4\uic\Loader\loader.py", line 71, in loadUi
  File "site-packages\PyQt4\uic\uiparser.py", line 982, in parse
  File "site-packages\PyQt4\uic\uiparser.py", line 803, in createUserInterface
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 478, in createLayout
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 521, in handleItem
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 265, in createWidget
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 265, in createWidget
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 478, in createLayout
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 521, in handleItem
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 265, in createWidget
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 478, in createLayout
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 521, in handleItem
  File "site-packages\PyQt4\uic\uiparser.py", line 781, in traverseWidgetTree
  File "site-packages\PyQt4\uic\uiparser.py", line 256, in createWidget
  File "site-packages\PyQt4\uic\uiparser.py", line 220, in setupObject
  File "site-packages\PyQt4\uic\objcreator.py", line 106, in createQObject
  File "site-packages\PyQt4\uic\objcreator.py", line 131, in findQObjectType
  File "site-packages\PyQt4\uic\Loader\qobjectcreator.py", line 107, in search
ImportError: No module named languagebox
```

File: buildscripts/builder.sh (hunk 1 above)
Comment (PeterSurda) [reply buildscripts/builder.sh]: line 145-148 show what needs to be installed.

**Comment by PeterSurda** [comment]
ok let me check the executable

**Comment by PeterSurda** [comment]
Ok I'm getting the same error, let me work on it.

**Comment by PeterSurda** [comment]
I think this is triggered by settings refactoring, the languagebox isn't imported in a .py file but in a .ui file so I have to add it as a hidden import, or find out how to auto-add .ui imports into the spec file.

**Comment by PeterSurda** [comment]
I manually added the hidden import into `bitmessagemain.spec` and it works now. I tested both the 32bit and 64bit version on Win7, and 32bit version on WindowsXP However as you continue with UI refactoring the issue could reappear elsewhere. For longer time it would be good to add a section in the spec file where it automatically gathers the imports from `.ui` files.

I'll try to see if I can get OpenCL patched, however I currently don't have any system available to test it on.

**Comment by PeterSurda** [comment]
Can't get it working on XP. I think that's for another PR.

**Comment by g1itch** [comment]
BTW: maybe move all scripts from `build` to `buildscripts` and add `build` to `.gitignore`? build is a special dir for setuptools.

**Comment by PeterSurda** [comment]
I agree, I'll make a separate commit for that.

**Pull request merged**